fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java

```diff
@@ -1601,6 +1605,23 @@ private void transferToMaster() {
         }
     }
 
+    /*
+     * Advance the id generator, ensuring it doesn't roll back.
+     *
+     * If we need to support time travel, the next id cannot be rolled back to avoid
+     * errors in the corresponding relationship of the metadata recorded in BE/MS.
+     */
+    void advanceNextId() {
+        long currentId = idGenerator.getBatchEndId();
+        long currentNanos = System.nanoTime();
+        long nextId = currentId + 1;
+        if (nextId < currentNanos) {
+            nextId = currentNanos;
+        }
+        idGenerator.setId(nextId);
```

Comment:
Do we need to persist the advanced id in case of wallclock rollback?
And, maybe there are some other IdGenerators that need to be taken into consideration.
<img width="416" alt="image" src="https://github.com/apache/doris/assets/6919662/074dd5f7-e8e4-475c-ac05-14d03d71e291">

Reply:
These id Generators are related to queries, not persistent meta.